occurrences/admin.py

```diff
@@ -34,23 +34,51 @@ class OccurrenceInline(admin.TabularInline):
     model = Occurrence.study_groups.through
     extra = 0
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    # Without setting these add and delete permissions to false, the page loads
+    # slowly due to a large number of SQL queries
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
 
 class EventQueueEnrolmentInline(admin.TabularInline):
     model = EventQueueEnrolment
     extra = 0
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    # Without setting these add and delete permissions to false, the page loads
+    # slowly due to a large number of SQL queries
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
 
 class EnrolmentInline(admin.TabularInline):
     model = Enrolment
     extra = 0
     exclude = ("person_deleted_at",)
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
```

Comment:
I think that here in `EnrolmentInline` this `has_change_permission` is actually fine, but I would still use the `autocomplete_fields` also when adding the enrolments, because without the search feature, the dropdown is just useless.

```suggestion
    autocomplete_fields = ['study_group', 'occurrence']
    
    def has_change_permission(self, request, obj=None):
        return False

```

Reply:
I added `autocomplete_fields` but removed `has_change_permission` because setting the `autocomplete_fields` fixed the original issue. It's not necessary to restrict editing capability here either.